src/service/resources-impl.js

```diff
@@ -1353,8 +1353,8 @@ export class Resources {
     }
 
     if (request) {
-      request.newHeight = newHeight;
-      request.newWidth = newWidth;
+      request.newHeight = newHeight || request.newHeight;
+      request.newWidth = newWidth || request.newWidth;
```

Comment:
Definitely. We also need to consider `newHeight = 0`, and that the combined might fail because it violates a heights rule but the width only should have succeeded.